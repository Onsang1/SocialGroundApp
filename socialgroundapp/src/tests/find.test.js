// /**
// * @jest-environment jsdom
// */

// import React from 'react';
// // import testing library functions
// import { BrowserRouter as Router } from 'react-router-dom';
// import { render, screen } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
// import renderer from 'react-test-renderer';
// import Feed from "../components/ActivityFeed";
// import Post from "../components/Post";
// import Comment from "../components/Comment";
// import CreateComment from '../components/Create';


// test('renders Send Button', () => {
//   render(
//     <Router>
//       <CreateComment />
//     </Router>,
//   ); 
//   const linkElement = getByText('SEND');
//   expect(linkElement).toBeInTheDocument();
// });


// test('renders username mario', () => {
//   render(
//     <Router>
//       <Post />
//     </Router>,
//   ); 
//   const linkElement = getByText('mario');
//   expect(linkElement).toBeInTheDocument();
// });

// test('renders username yoshi', () => {
//   render(
//     <Router>
//       <Post />
//     </Router>,
//   ); 
//   const linkElement = screen.getByText('yoshi');
//   expect(linkElement).toBeInTheDocument();
// });

// test('renders username yoshi', () => {
//   render(
//     <Router>
//       <Comment />
//     </Router>,
//   ); 
//   const linkElement = screen.getByText('username');
//   expect(linkElement).toBeInTheDocument();
// });

// test('feed matches snapshot', () => {
//   const component = renderer.create(
//     <Router>
//       <Feed />
//     </Router>,
//   );
//   const tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });

// test('comment matches snapshot', () => {
//   const component = renderer.create(
//     <Router>
//       <Comment />
//     </Router>,
//   );
//   const tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });


// test('Textbox to have value after input', async () => {
//   render(
//     <Router>
//       <CreateComment />
//     </Router>,
//   );

//   const commentLabel = getByText('Type');
//   const commentPlaceholder = getByText('Add a comment');

//   expect(commentLabel).toBeInDocument();
//   expect(commentPlaceholder).toBeInDocument();

//   const commentInput = getByLabelText('Add a comment');

//   expect(commentInput).toHaveAttribute('maxRows', 'placeholder','required','variant');

//   const element = screen.getByRole('textbox', { name: 'Add a comment' });

//   await userEvent.type(element, 'username123@gmail.com');
//   expect(screen.getByLabelText('Add a comment')).toHaveValue('user123@gmail.com');
// });

// test('Should call onSubmit when Send button is clicked', async () => {
//   render(
//     <Router>
//       <CreateComment />
//     </Router>,
//   );
//   await userEvent.type(screen.getByRole('textbox', { name: 'Add a comment' }), 'user123@gmail.com');
//   userEvent.click(screen.getByRole('button', { name: 'SEND' }));
//   expect(onsubmit).toHaveBeenCalledWith('user123@gmail.com');
// });
