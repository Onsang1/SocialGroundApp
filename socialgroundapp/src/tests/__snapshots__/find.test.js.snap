// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Find matches snapshot 1`] = `
Array [
  <div
    className="Find-Page"
  >
    <div
      className="logo-area"
    >
      <a
        href="/Feed"
        onClick={[Function]}
      >
        <button
          className="logo-button"
          type="button"
        >
          <img
            alt="large_logo"
            className="large-logo"
            src={Object {}}
          />
        </button>
      </a>
    </div>
    <h3
      className="FindMessage"
    >
      Find Friends on Social Ground
    </h3>
    <div
      className="search-area"
    >
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="outlined-basic"
          id="outlined-basic-label"
        >
          Search
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="outlined-basic"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter username"
            required={false}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-yjsfm1"
            >
              <span>
                Search
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <p
      className="suggestedText"
    >
       Suggestions
    </p>
  </div>,
  ",",
]
`;

exports[`NotFound matches snapshot 1`] = `
Array [
  <div
    className="not-found"
  >
    <h2>
      Sorry
    </h2>
    <p>
      That page cannot be found
    </p>
    <a
      href="/Feed"
      onClick={[Function]}
    >
      Back to the homepage...
    </a>
  </div>,
  ",",
]
`;
